lib/widgets/emoji_reaction.dart

```diff
@@ -0,0 +1,264 @@
+                      case ReactionType.unicodeEmoji:
+                        return _UnicodeEmoji(
+                          emojiCode: emojiCode,
+                          emojiName: emojiName,
+                          selected: selfVoted,
+                        );
+                      case ReactionType.realmEmoji:
+                      case ReactionType.zulipExtraEmoji:
+                        return _ImageEmoji(
+                          emojiCode: emojiCode,
+                          emojiName: emojiName,
+                          selected: selfVoted,
+                        );
+                    }
+                  })()),
+                Flexible(
+                  // Added vertical: 1 to give some space when the label is
+                  // taller than the emoji (e.g. because it needs multiple lines)
+                  child: Padding(padding: const EdgeInsets.symmetric(horizontal: 3, vertical: 1),
+                    child: Text(
+                      textWidthBasis: TextWidthBasis.longestLine,
+                      style: TextStyle(
+                        fontFamily: 'Source Sans 3',
+                        fontSize: (14 * 0.90),
+                        height: 13 / (14 * 0.90),
+                        color: labelColor,
+                      ).merge(selfVoted
+                        ? weightVariableTextStyle(context, wght: 600, wghtIfPlatformRequestsBold: 900)
+                        : weightVariableTextStyle(context)),
+                      label)),
+                ),
+              ])))));
+  }
+}
+
+class _UnicodeEmoji extends StatelessWidget {
+  const _UnicodeEmoji({
+    required this.emojiCode,
+    required this.emojiName,
+    required this.selected,
+  });
+
+  final String emojiCode;
+  final String emojiName;
+  final bool selected;
+
+  @override
+  Widget build(BuildContext context) {
+    final parsed = tryParseEmojiCodeToUnicode(emojiCode);
+    if (parsed == null) {
+      return _TextEmoji(emojiName: emojiName, selected: selected);
+    }
+    final textScaler = MediaQuery.textScalerOf(context);
+    return SizedBox(
+      width: textScaler.scale(17),
+      child: Text(
+        style: const TextStyle(fontSize: 17),
+        strutStyle: const StrutStyle(fontSize: 17, forceStrutHeight: true),
+        parsed));
+  }
+}
+
+class _ImageEmoji extends StatelessWidget {
+  const _ImageEmoji({
+    required this.emojiCode,
+    required this.emojiName,
+    required this.selected,
+  });
+
+  final String emojiCode;
+  final String emojiName;
+  final bool selected;
+
+  Widget get _textFallback => _TextEmoji(emojiName: emojiName, selected: selected);
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+
+    // Some people really dislike animated emoji.
+    final doNotAnimate = MediaQuery.of(context).disableAnimations
+      || PlatformDispatcher.instance.accessibilityFeatures.reduceMotion;
```

Comment:
Yes, these are two different settings. See `AccessibilityFeatures.reduceMotion` and `AccessibilityFeatures.disableAnimations`: https://api.flutter.dev/flutter/dart-ui/AccessibilityFeatures-class.html

It says "reduce motion" is an iOS-only setting, though. Probably best/safest to only check it on iOS; I'll do that.

"Disable animations" is exposed on `MediaQueryData`: https://api.flutter.dev/flutter/widgets/MediaQueryData/disableAnimations.html

But "reduce motion" is not. There is an issue to expose it there (https://github.com/flutter/flutter/issues/65874), but it looks like something else might be done instead; see details there.

Reply:
Cool, thanks.

I think "reduce motion" probably shouldn't control this. Here's what looks to be Apple dev docs for it:
https://developer.apple.com/documentation/swiftui/environmentvalues/accessibilityreducemotion
> If this property’s value is true, UI should avoid large animations, especially those that simulate the third dimension.

And the user-facing description, from the screenshot in that upstream image, is:
> Reduce the motion of the user interface, including the parallax effect of icons.

So that sounds like a different type of thing from this.

Reply:
We discussed this further in our call today, and it looks like `disableAnimations` actually never gets set on iOS. Given that, we should probably consult `reduceMotion` after all (like the current revision does) so that the user has some way to ask the animations to stop.

Then recent versions of iOS (iOS 17+, so new this year) do have a more closely relevant setting:
https://developer.apple.com/documentation/swiftui/environmentvalues/accessibilityplayanimatedimages
which is documented for users at
https://support.apple.com/guide/iphone/reduce-onscreen-motion-iph0b691d3ed/ios
as:
> _Auto-Play Animated Images:_ […] (When on, rapid animated images and moving elements such as GIFs in Messages and Safari play automatically.)

So there should be a Flutter issue upstream to provide that value, either by having it control `disableAnimations` or by adding a new separate field on `AccessibilityFeatures`.

If there's such an issue already, we can link to it in a comment here. If not, we can leave a TODO comment for ourselves to file that issue — though actually filing it would be a task for after Beta 1.

Reply:
Cool; yeah. Thanks for writing that down!